code/modules/uplink/uplink_items/device_tools.dm

```diff
@@ -0,0 +1,227 @@
+		return
+	uplink.failsafe_code = uplink.generate_code()
+	var/code = "[islist(uplink.failsafe_code) ? english_list(uplink.failsafe_code) : uplink.failsafe_code]"
+	to_chat(user, span_warning("The new failsafe code for this uplink is now : [code]. You may check your antagonist info to recall this."))
+	return source //For log icon
+
+/datum/uplink_item/device_tools/toolbox
+	name = "Full Syndicate Toolbox"
+	desc = "The Syndicate toolbox is a suspicious black and red. It comes loaded with a full tool set including a \
+			multitool and combat gloves that are resistant to shocks and heat."
+	item = /obj/item/storage/toolbox/syndicate
+	cost = 1
+	illegal_tech = FALSE
+
+/datum/uplink_item/device_tools/rad_laser
+	name = "Radioactive Microlaser"
+	desc = "A radioactive microlaser disguised as a standard Nanotrasen health analyzer. When used, it emits a \
+			powerful burst of radiation, which, after a short delay, can incapacitate all but the most protected \
+			of humanoids. It has two settings: intensity, which controls the power of the radiation, \
+			and wavelength, which controls the delay before the effect kicks in."
+	item = /obj/item/healthanalyzer/rad_laser
+	cost = 3
+
+
+/datum/uplink_item/device_tools/suspiciousphone
+	name = "Protocol CRAB-17 Phone"
+	desc = "The Protocol CRAB-17 Phone, a phone borrowed from an unknown third party, it can be used to crash the space market, funneling the losses of the crew to your bank account.\
+	The crew can move their funds to a new banking site though, unless they HODL, in which case they deserve it."
+	item = /obj/item/suspiciousphone
+	restricted = TRUE
+	cost = 7
+	limited_stock = 1
+
+/datum/uplink_item/device_tools/binary
+	name = "Binary Translator Key"
+	desc = "A key that, when inserted into a radio headset, allows you to listen to and talk with silicon-based lifeforms, \
+			such as AI units and cyborgs, over their private binary channel. Caution should \
+			be taken while doing this, as unless they are allied with you, they are programmed to report such intrusions."
+	item = /obj/item/encryptionkey/binary
+	cost = 5
+	surplus = 75
+	restricted = TRUE
+
+// Low progression cost
+
+/datum/uplink_item/device_tools/emag
+	name = "Cryptographic Sequencer"
+	desc = "The cryptographic sequencer, electromagnetic card, or emag, is a small card that unlocks hidden functions \
+			in electronic devices, subverts intended functions, and easily breaks security mechanisms. Cannot be used to open airlocks."
+	progression_minimum = 20 MINUTES
+	item = /obj/item/card/emag
+	cost = 4
+
+/datum/uplink_item/device_tools/stimpack
+	name = "Stimpack"
+	desc = "Stimpacks, the tool of many great heroes, make you nearly immune to stuns and knockdowns for about \
+			5 minutes after injection."
+	progression_minimum = 20 MINUTES
+	item = /obj/item/reagent_containers/hypospray/medipen/stimulants
+	cost = 5
+	surplus = 90
+
+
+// Medium progression cost
+
+/datum/uplink_item/device_tools/hacked_module
```

Comment:
This is so late as to be entirely pointless. The work to get an upload early is what made this so formidable. Once you get into later parts of the round, the upload can be printed, and so can just use a freeform module or one human module. I've never been a fan of the hacked module due to redundancy, but this feels especially true seeing it only available after significant rep gain/30 minute wait period.